src/infra/core/src/repos/metadata_chain_impl.rs

```diff
@@ -720,14 +565,18 @@ where
         opts: AppendOpts<'a>,
     ) -> Result<Multihash, AppendError> {
         if opts.validation == AppendValidation::Full {
-            self.validate_append_prev_block_exists(&block).await?;
-            self.validate_append_sequence_numbers_integrity(&block)
-                .await?;
-            self.validate_append_seed_block_order(&block)?;
-            self.validate_append_system_time_is_monotonic(&block)
-                .await?;
-            self.validate_append_watermark_is_monotonic(&block).await?;
-            self.validate_append_offsets_are_sequential(&block).await?;
+            let validators: BoxedVisitors = vec![
+                Box::<ValidateSeedBlockOrderVisitor>::default(),
+                Box::<ValidatePrevBlockExistsVisitor>::default(),
+                Box::<ValidateSequenceNumbersIntegrityVisitor>::default(),
+                Box::<ValidateSystemTimeIsMonotonicVisitor>::default(),
+                Box::<ValidateWatermarkIsMonotonicVisitor>::default(),
+                Box::<ValidateOffsetsAreSequentialVisitor>::default(),
```

Comment:
Do you think it's better to box them or perhaps we could pass them all as `&[&mut dyn Visitor]`?

Reply:
Yes, I have endeavored to do so -- my past attempts have been unsuccessful. 

But now, I've been able to reduce the number of allocations, at least at this point.

Reply:
It is likely that all allocations related to Visitors have been eliminated